Sources/Nimble/Adapters/AssertionRecorder.swift

```diff
@@ -42,8 +42,11 @@ public class AssertionRecorder: AssertionHandler {
 ///
 /// Once the closure finishes, then the original Nimble assertion handler is restored.
 ///
+/// TEMP: @escaping annotation to closure param is workaround to Radar 40857699
+/// https://openradar.appspot.com/radar?id=5595735974215680
+///
 /// @see AssertionHandler
-public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: () throws -> Void) {
+public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: @escaping () throws -> Void) {
```

Comment:
Promoting a closure to `@escaping` will make Swift require explicit `self.` when accessing properties from within that closure, making this a breaking change.

While this isn't the same as `#if XCODE_10`, could the following be a compromise?

```
#if swift(>=4.2)
public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: @escaping () throws -> Void) {
#else
public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: () throws -> Void) {
#endif
``

Reply:
Using `withoutActuallyEscaping` may be another option.

```diff
diff --git a/Sources/Nimble/Adapters/AssertionRecorder.swift b/Sources/Nimble/Adapters/AssertionRecorder.swift
index 1aad65a..0f7adbc 100644
--- a/Sources/Nimble/Adapters/AssertionRecorder.swift
+++ b/Sources/Nimble/Adapters/AssertionRecorder.swift
@@ -46,15 +46,17 @@ public class AssertionRecorder: AssertionHandler {
 /// https://openradar.appspot.com/radar?id=5595735974215680
 ///
 /// @see AssertionHandler
-public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: @escaping () throws -> Void) {
+public func withAssertionHandler(_ tempAssertionHandler: AssertionHandler, closure: () throws -> Void) {
     let environment = NimbleEnvironment.activeInstance
     let oldRecorder = environment.assertionHandler
     let capturer = NMBExceptionCapture(handler: nil, finally: ({
         environment.assertionHandler = oldRecorder
     }))
     environment.assertionHandler = tempAssertionHandler
-    capturer.tryBlock {
-        try! closure()
+    withoutActuallyEscaping(closure) { escapable in
+        capturer.tryBlock {
+            try! escapable()
+        }
     }
 }
```